### PR TITLE
API endpoint to get all the submissions made by user for a challenge phase.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
#421 solved. @deshraj Please review the PR.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -6,13 +6,14 @@
 # C1001: Old-style class defined.
 # D100:  Missing docstring in public module.
 # E1101: %s %r has no %r member.
+# MC0001: mccabe plugin
```
Comment (deshraj) [inline .pylintrc]: @RishabhJain2018 this file should not be included in this PR? Btw, why do you need this?

File: apps/jobs/views.py [hunk 2]
```diff
@@ -18,10 +19,11 @@
     get_participant_team_id_of_user_for_a_challenge,)
 
 from .serializers import SubmissionSerializer
+from .models import Submission
```
Comment (deshraj) [inline apps/jobs/views.py]: Can you make swap line 21 and 22?

File: apps/jobs/views.py [hunk 3]
```diff
@@ -61,13 +63,22 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
...
         return Response(response_data, status=status.HTTP_403_FORBIDDEN)
 
-    serializer = SubmissionSerializer(data=request.data,
-                                      context={'participant_team': participant_team,
-                                               'challenge_phase': challenge_phase,
-                                               'request': request
-                                               })
-    if serializer.is_valid():
-        serializer.save()
+    if request.method == 'GET':
```
Comment (deshraj) [inline apps/jobs/views.py]: `GET` method should be at the start of the view since most of these checks doesn't apply for the `GET` request.

File: apps/jobs/views.py [hunk 4]
```diff
@@ -61,13 +63,22 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
...
+        submission = Submission.objects.filter(participant_team=participant_team_id,
+                                               challenge_phase=challenge_phase)
+        paginator, result_page = paginated_queryset(submission, request)
+        serializer = SubmissionSerializer(result_page, many=True)
```
Comment (deshraj) [inline apps/jobs/views.py]: Check for serializer errors in `GET` method as well.

File: tests/unit/jobs/test_views.py [hunk 5]
```diff
@@ -258,3 +259,130 @@ def test_challenge_submission_for_successful_submission(self):
         response = self.client.post(self.url, {
                                     'status': 'submitting', 'input_file': self.input_file}, format="multipart")
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class GetChallengeSubmittedTest(BaseAPITestClass):
```
Comment (deshraj) [inline tests/unit/jobs/test_views.py]: You would need to modify the tests also when you will modify the view.

File: .pylintrc (hunk 1 above)
Comment (RishabhJain2018) [inline .pylintrc]: @deshraj Yeah, now we don't need this. I just tried to disable a pylint plugin `mccabe`. I have done that using the `.landscape.yml` file.

File: apps/jobs/views.py (hunk 2 above)
Comment (RishabhJain2018) [inline apps/jobs/views.py]: yeah sure.

File: tests/unit/jobs/test_views.py (hunk 5 above)
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: yeah sure.

File: apps/jobs/urls.py [hunk 6]
```diff
@@ -6,4 +6,7 @@
     url(r'challenge/(?P<challenge_id>[0-9]+)/'
         r'challenge_phase/(?P<challenge_phase_id>[0-9]+)/submission/',
         views.challenge_submission, name='challenge_submission'),
+    url(r'get_challenge/(?P<challenge_id>[0-9]+)/'
```
Comment (deshraj) [inline apps/jobs/urls.py]: @TroJan suggestions for the URL here?

File: apps/jobs/views.py [hunk 7]
```diff
@@ -71,3 +73,47 @@ def challenge_submission(request, challenge_id, challenge_phase_id):
...
+    # check if the challenge phase exists or not
+    try:
+        challenge_phase = ChallengePhase.objects.get(
+            pk=challenge_phase_id, challenge=challenge)
+    except ChallengePhase.DoesNotExist:
+        response_data = {'error': 'Challenge Phase does not exist'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    # getting participant team object for the user for a particular challenge.
+    participant_team_id = get_participant_team_id_of_user_for_a_challenge(
```
Comment (deshraj) [inline apps/jobs/views.py]: @RishabhJain2018 this should come in try except block.

Comment (deshraj) [inline apps/jobs/views.py]: Sorry, my bad. This looks good.

File: tests/unit/jobs/test_views.py [hunk 8]
```diff
@@ -258,3 +259,99 @@ def test_challenge_submission_for_successful_submission(self):
...
+        # Note that we haven't added the self.participant_team to Challenge
+        expected = {
+            'error': 'You haven\'t participated in the challenge'
+        }
+
+        response = self.client.get(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_get_challenge_submitted(self):
```
Comment (deshraj) [inline tests/unit/jobs/test_views.py]: Can you rename this to `test_get_challenge_submissions`?

File: tests/unit/jobs/test_views.py [hunk 9]
```diff
@@ -258,3 +259,99 @@ def test_challenge_submission_for_successful_submission(self):
...
+                'challenge_phase': self.submission.challenge_phase.pk,
+                'created_by': self.submission.created_by.pk,
+                'status': self.submission.status,
+                'input_file': self.submission.input_file
+            }
+        ]
+        self.challenge.participant_teams.add(self.participant_team)
+        self.challenge.save()
+        response = self.client.get(self.url, {})
+        print "response********", response.data
```
Comment (deshraj) [inline tests/unit/jobs/test_views.py]: please remove the debug statement.

File: tests/unit/jobs/test_views.py [hunk 10]
```diff
@@ -258,3 +259,98 @@ def test_challenge_submission_for_successful_submission(self):
         response = self.client.post(self.url, {
                                     'status': 'submitting', 'input_file': self.input_file}, format="multipart")
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class GetChallengeSubmittedTest(BaseAPITestClass):
```
Comment (deshraj) [inline tests/unit/jobs/test_views.py]: Please rename this to `GetChallengeSubmissionTest`

File: tests/unit/jobs/test_views.py (hunk 8 above)
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: okk sure.

File: tests/unit/jobs/test_views.py (hunk 10 above)
Comment (RishabhJain2018) [inline tests/unit/jobs/test_views.py]: yeah sure.

**Comment by deshraj** [comment]
Looks good. :)
